server/services/authorization/service.ts

```diff
@@ -83,6 +89,38 @@ export class Authorization extends ServiceClass implements IAuthorization {
 		AuthorizationUtils.addRolePermissionWhiteList(role, permissions);
 	}
 
+	async getUsersFromPublicRoles(): Promise<Pick<IUser, '_id' | 'username' | 'roles'>[]> {
+		const roleIds = await this.getPublicRoles();
+
+		return this.getUserFromRoles(roleIds);
+	}
+
+	private getPublicRoles = mem(async (): Promise<string[]> => {
+		const roles = await this.Roles.find({ scope: 'Users', description: { $exists: 1, $ne: '' } }, { projection: { _id: 1 } }).toArray();
+
+		return roles.map(({ _id }) => _id);
+	}, { maxAge: 10000 });
+
+	private getUserFromRoles = mem(async (roleIds: string[]) => {
+		const options = {
+			sort: {
+				username: 1,
+			},
+			fields: {
+				username: 1,
+				roles: 1,
+			},
+		};
+
+		const users = await this.Users.findUsersInRoles(roleIds, null, options).toArray();
```

Comment:
If there happens to be a role all users have, this query may return the whole list of users (which can be quite big). Any plans to paginate this? 👀

Reply:
I see :thinking: I didn't change the previous behavior though.. maybe we should deprecate this method :thinking:

Reply:
Yeah 🤔 that would be a good thing

Reply:
Agreed, I'll create a task to deprecate this :+1: